### ISSUE TITLE
[libprotobuf FATAL google/protobuf/map_field.cc:364] CHECK failed: IsMapValid() && other.IsMapValid()

### ISSUE THREAD

**Issue by Paperone80** [protocolbuffers/protobuf#9887]
**What version of protobuf and what language are you using?**
Version: 3.20.1
Language: Python3.9

**What operating system (Linux, Windows, ...) and version?**
Linux 2b4369a98013 4.15.0-175-generic 16.04.1-Ubuntu SMP Fri Mar 25 10:27:57 UTC 2022 x86_64 GNU/Linux

**What runtime / compiler are you using (e.g., python version or gcc version)**
3.9

**What did you do?**
repro_signature_bug.py
```
#!/usr/bin/env python3

"""Reproduce a crash in libprotobuf."""

import os, sys, io, pathlib, json

import google

import tensorflow as tf

from tensorflow_serving.apis import get_model_metadata_pb2
from tensorflow.python.tools import saved_model_utils, saved_model_cli

from google.protobuf import json_format

print("Google Protobuf v{}".format(google.protobuf.__version__))
print("Tensorflow v{}".format(tf.__version__))

def load_model(model_name):
  base_url = 'http://download.tensorflow.org/models/object_detection/'
  model_file = model_name + '.tar.gz'
  model_dir = tf.keras.utils.get_file(
    fname=model_name, 
    origin=base_url + model_file,
    untar=True)

  model_dir = str(pathlib.Path(model_dir)/"saved_model")

  return str(model_dir)


def get_signature_dict(model_dir, make_it_work: bool=True):

  # Get the 'meta_graph_def'
  meta_graph_def = saved_model_utils.get_meta_graph_def(
      str(model_dir), tag_set='serve'
  )

  # Get the 'signature_def' from 'meta_graph_def'
  signature_def_map = meta_graph_def.signature_def
  assert isinstance(signature_def_map, google.protobuf.pyext._message.MessageMapContainer)

  if make_it_work:
    dummy = list(signature_def_map)
  else:
    pass

  # Create an empty 'model_signature_def_map' with .DESCRIPTOR
  model_signature_def_map = get_model_metadata_pb2.SignatureDefMap()
  assert isinstance(model_signature_def_map.signature_def, google.protobuf.pyext._message.MessageMapContainer)

  # Merge 'signature_def_map' onto empty 'model_signature_def_map.signature_def'
  model_signature_def_map.signature_def.MergeFrom(signature_def_map)

  # Now convert to dict()
  signature_def_dict = json.loads(
    json_format.MessageToJson(model_signature_def_map)
  )

  return signature_def_dict

if __name__ == "__main__":
  model_name = 'ssd_mobilenet_v1_coco_2017_11_17'
  model_dir = load_model(model_name)

  print("\nStarting 'make_it_work=True':\n", get_signature_dict(model_dir, make_it_work=True))

  print("\nStarting 'make_it_work=False'...\n")
  print(get_signature_dict(model_dir, make_it_work=False))
```
Dockerfile
```
FROM python:3.9

RUN python3 -m pip install protobuf grpcio-tools
RUN python3 -m pip install --upgrade tensorflow=="2.*" tensorflow-serving-api

COPY . /t
WORKDIR /t

ENV PYTHONUNBUFFERED=1

RUN uname -a

RUN PYTHONUNBUFFERED=1 python3 repro_signature_bug.py
```
Build docker 
`sudo docker build -f Dockerfile .`

**What did you expect to see**
Tensorflow Model signature def as dictionary.
 `{'signatureDef': {'serving_default': {'inputs': {'inputs': {'name': 'image_tensor:0', 'dtype': 'DT_UINT8', 'tensorShape': {'dim': [{'size': '-1'}, {'size': '-1'}, {'size': '-1'}, {'size': '3'}]}}}, 'outputs': {'detection_scores': {'name': 'detection_scores:0', 'dtype': 'DT_FLOAT', 'tensorShape': {'dim': [{'size': '-1'}, {'size': '100'}]}}, 'detection_classes': {'name': 'detection_classes:0', 'dtype': 'DT_FLOAT', 'tensorShape': {'dim': [{'size': '-1'}, {'size': '100'}]}}, 'num_detections': {'name': 'num_detections:0', 'dtype': 'DT_FLOAT', 'tensorShape': {'dim': [{'size': '-1'}]}}, 'detection_boxes': {'name': 'detection_boxes:0', 'dtype': 'DT_FLOAT', 'tensorShape': {'dim': [{'size': '-1'}, {'size': '100'}, {'size': '4'}]}}}, 'methodName': 'tensorflow/serving/predict'}}}
`

**What did you see instead?**

```
[libprotobuf FATAL google/protobuf/map_field.cc:364] CHECK failed: IsMapValid() && other.IsMapValid(): 
terminate called after throwing an instance of 'google::protobuf::FatalException'
  what():  CHECK failed: IsMapValid() && other.IsMapValid(): 
Aborted (core dumped)
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

**Anything else we should know about your project / environment**
Compiling latest Protobuf 3.20.1 from source avoids the fatal error but returns an empty 'signature_def_dict' = {}.

**Comment by Paperone80** [comment]
@coryan @fowles Found #9463 so I compiled from source but did not resolve the issue.

**Comment by anandolee** [comment]
#9463 is a known issue for our ParseFromString() for map in c++. I will see what I can do for it.

This one I am not sure what caused the issue. I saw there's a `make_it_work'. Am I right the error only happens when `make_it_work` is true, and the merge works well when `make_it_work' is flase?
